iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergMetaHook.java

```diff
@@ -250,4 +312,11 @@ private static PartitionSpec spec(Schema schema, Properties properties,
       return PartitionSpec.unpartitioned();
     }
   }
+
+  private static class PreAlterTableProperties {
+    private static String tableLocation;
```

Comment:
Why do we do this? What happens if we have 2 parallel alterTable-s?

Reply:
Why make the fields static? I also think it could create concurrency issues. You could create this as an object similarly to how the catalogProperties are currently cached in the MetaHook between pre and post executions

Reply:
Frankly, I don't know why I left it as static :). Good catch!